packages/editor/src/components/page-attributes/parent.js

```diff
@@ -1,44 +1,195 @@
+	 * @param {Function} populateResults A callback function which receives the results.
+	 */
+	suggestPage( query, populateResults ) {
+		const { items } = this.props;
+
+		if ( query === items ) {
+			populateResults( this.resolveResults( items ) );
+			return;
+		}
+
+		if ( query.length < 2 ) {
+			populateResults( this.resolveResults( items ) );
+			return;
+		}
+
+		if ( this.searchCache[ query ] ) {
+			populateResults( this.resolveResults( this.searchCache[ query ] ) );
+			return;
+		}
+
+		this.requestResults( query, populateResults );
+	}
+
+	render() {
+		const { postType, items, onUpdateParent, parent } = this.props;
+		const { parentPost } = this.state;
+		let currentParent = false;
+
+		if ( ! parentPost && false !== parent ) {
+			if ( 0 === parent ) {
+				currentParent = '';
+			} else {
+				// We have the parent page id, we need to display its name.
+				const currentParentFromItems = items && items.find( ( item ) => {
+					return item.id === parent;
+				} );
+
+				// Set or fetch the current author.
+				if ( currentParentFromItems ) {
+					this.setState( {
+						parentPost: parent,
+					} );
+				} else {
+					this.getCurrentParentFromAPI( parent );
+				}
+			}
+		}
+
+		const isHierarchical = get( postType, [ 'hierarchical' ], false );
+		const parentPageLabel = get( postType, [ 'labels', 'parent_item_colon' ] );
+		const pageItems = items || [];
+		if ( ! isHierarchical || ! parentPageLabel || ! pageItems.length ) {
+			return null;
+		}
+
+		if ( false === currentParent ) {
+			currentParent = parentPost && parentPost.title.rendered ?
+				`${ parentPost.title.rendered } (#${ parentPost.id })` :
+				`${ __( 'no title' ) } (#${ parentPost.id })`;
+		}
+
+		if ( items.length > 99 ) {
+			return (
+				<>
+					<label htmlFor={ parent }>{ __( 'Parent Page' ) }</label>
+					<Autocomplete
+						id={ parent }
+						minLength={ 2 }
+						showAllValues={ true }
+						defaultValue={ currentParent ? currentParent : '' }
+						displayMenu="overlay"
+						onConfirm={ this.handleSelection }
+						source={ this.suggestPage }
+						showNoOptionsFound={ false }
+						preserveNullOptions={ true }
+						tStatusQueryTooShort={ ( minQueryLength ) =>
+							// translators: %d: the number characters required to initiate a page search.
+							sprintf( __( 'Type in %d or more characters for results' ), minQueryLength )
+						}
+						tStatusNoResults={ () => __( 'No search results' ) }
+						// translators: 1: the index of thre selected result. 2: The total number of results.
+						tStatusSelectedOption={ ( selectedOption, length ) => sprintf( __( '%1$s (1 of %2$s) is selected' ), selectedOption, length ) }
+						tStatusResults={ ( length, contentSelectedOption ) => {
+							return (
+								_n( '%d result is available.', '%d results are available.', length ) +
+								' ' + contentSelectedOption
+							);
+						} }
+						cssNamespace="components-parent-page__autocomplete"
```

Comment:
It should be `editor-page-attributes__parent`.

>https://developer.wordpress.org/block-editor/contributors/develop/coding-guidelines/

Reply:
👍